### PR TITLE
implemented filtering of Questions feature

### PR DESCRIPTION
**Pull request by Sandipan-Ghosh**
**Fixed Issues**

#21451 #21452 #21453 #21454 #21455 #21456

**Files Changed**

* question-dashboard.component.ts 
* question-dashboard.html

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [x] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [x] All test cases are executed provided by tester.

**Known Issues**

N/A

**Comments**

Filtering and Searching feature is implemented

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 1]
```diff
@@ -19,13 +19,9 @@ public QuestionRepository(TrappistDbContext dbContext)
...
 
-        /// <summary>
-        /// Method to get all the Questions
-        /// </summary>
-        /// <returns>Question list</returns>
-        public async Task<ICollection<Question>> GetAllQuestionsAsync()
+        public async Task<ICollection<Question>> GetAllQuestionsAsync(string userId)
         {
-            return (await _dbContext.Question.Include(x => x.Category).Include(x => x.CodeSnippetQuestion).Include(x => x.SingleMultipleAnswerQuestion).ThenInclude(x => x.SingleMultipleAnswerQuestionOption).OrderByDescending(g => g.CreatedDateTime).ToListAsync());
+            return (await _dbContext.Question.Include(x => x.Category).Include(x => x.CodeSnippetQuestion).Include(x => x.SingleMultipleAnswerQuestion).ThenInclude(x => x.SingleMultipleAnswerQuestionOption).OrderByDescending(g => g.CreatedDateTime).Where(u=>u.CreatedByUserId.Equals(userId)).ToListAsync());
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: first, filter the question table then include foreign key associated tables.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts [hunk 2]
```diff
@@ -51,11 +69,56 @@ export class QuestionsDashboardComponent implements OnInit {
     //To get all the Questions
     getAllQuestions() {
         this.questionsService.getQuestions().subscribe((questionsList) => {
-            this.questionDisplay = questionsList;
+            this.question = questionsList;
+            this.selectedCategory = new Category;
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts]: why brackets are missing?

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts [hunk 3]
```diff
@@ -95,4 +158,60 @@ export class QuestionsDashboardComponent implements OnInit {
...
+    /**
+     * To count the number of Questions
+     */
+    countQuestion() {
+        this.easy = this.medium = this.hard = 0;
+        let questionList = this.filterQuestion(this.question, this.selectedCategory, null, this.matchString);
+        questionList.forEach(x => {
+            switch (x.difficultyLevel) {
+                case 0:
+                    return this.easy++;
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts]: what is the significance of return statement here?

Comment (Sandipan-Ghosh) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts]: returns the incremented value of this.easy whenever case 0 i.e. difficultyLevel easy is true.

Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/questions-dashboard/questions-dashboard.component.ts]: not able to understand that if you are returning to one condition then why foreach is required for it?

**Pull request merged**